base/spec/lib/stellar/sep10_spec.rb

```diff
@@ -187,12 +185,12 @@
       client_kp_b = Stellar::KeyPair.random
       client_kp_c = Stellar::KeyPair.random
       timeout = 600
-      anchor_name = "SDF"
+      domain = "www.stellar.org"
```

Comment:
This is not a fully qualified domain name. That would be `stellar.org` or `testanchor.stellar.org`.

Reply:
I believe technically, all there of those are FQDNs because they unambiguously identify the host (or service) in the DNS hierarchy. Strictly speaking FQDN has to end with the dot, which anchors the name to the root zone and plays pretty much the same role as the first slash in an absolute file path. It's often defined as the full name of the system, as opposed to just a hostname, so in classical terms `www.stellar.org` is FQDN (host named `www` under domain `stellar.org`).

But that's all terminology debate and I think more important here the fact that SDF's stellar.toml is actually available from `www.stellar.org/.well-known/stellar.toml`, and non-www version redirects to it.

Reply:
That first paragraph is enlightening, thank you for clarifying that. I tried to find better definitions than what I found by googling but never truly understood the format. @leighmcculloch we were just discussing this.

You implied that FQDN is actually a loosely-defined term, or that FQDN is a broader umbrella than we had thought. I think we need to harden the format specification for `home_domain`/`homeDomain` values in challenges and sent in API requests so we don't end up with incompatible client-anchor pairs.

> I think more important here the fact that SDF's stellar.toml is actually available from www.stellar.org/.well-known/stellar.toml

The `home_domain`/`homeDomain` value is the FQDN of the service requiring authentication. This could be a completely distinct domain from the TOML FQDN.

I'm personally responsible for that confusion by making the following statement in the issue description:

> client applications must match the value of the Manage Data operation's key with the home domain used in the request for the service's TOML file

This is incorrect, because I made the assumption that the SEP-24 or whatever Stellar service would have the same FQDN as the TOML, which is the case with the server I manage, `testanchor.stellar.org`.

Reply:
@leighmcculloch I might need to make a post to the stellar-dev group clarifying the above.

Reply:
@JakeUrban FQDN is certainly an ambiguous term. The most adequate "official" explanation I'm aware of is given in [RFC8499](https://tools.ietf.org/html/rfc8499#page-8)

> Fully-Qualified Domain Name (FQDN):  This is often just a clear way of saying the same thing as "domain name of a node", as outlined above.  However, the term is ambiguous.  Strictly speaking, a fully-qualified domain name would include every label, including the zero-length label of the root: such a name would be written `www.example.net.` (note the terminating dot).  But, because every name eventually shares the common root, names are often written relative to the root (such as `www.example.net`) and are still called "fully qualified".  This term first appeared in [RFC819](https://tools.ietf.org/html/rfc819). In this document, names are often written relative to the root.
>
> The need for the term "fully-qualified domain name" comes from the existence of partially qualified domain names, which are names where one or more of the last labels in the ordered list are omitted (for example, a domain name of "www" relative to "example.net" identifies "www.example.net").  Such relative names are understood only by context.

Another reference point could be `hostname` command from standard linux distro:

```
~ hostname
xlm
~ hostname --fqdn
xlm.do.nebols.in
~ hostname --domain
do.nebols.in
```

But I don't think that FQDN is the best term for our current task at hand, because... and here we get to confusing part.

I personally always thought of this FQDN as Stellar's `home_domain`, which implies that every web service implementing Stellar authentication would have its own `.well-known/stellar.toml`. And I am still not convinced  it should be something different, because that's suggested by the flow itself. 

Client generally starts from the state `( service_uri, keypair )` — wants to authenticate himself for service at `service_uri` with `keypair`. In order to do it, it's necessary to get service's `web_auth_endpoint` and `signing_keypair`, so it has to consult the TOML by making a request to `https://#{service_uri}/.well-known/stellar.toml`. It then starts the auth procedure with `web_auth_endpoint` by requesting the challenge, passing the tuple `( keypair.public_key, service_uri )`.

From that it's logical that "FQDN" is in fact URI of the service which we're authenticating with, with `$scheme://` part omitted because it's always https. Does that make sense or am I missing something here?